pkg/cli/mirror/copy.go

```diff
@@ -0,0 +1,594 @@
+	}
+	manifest, err := manifest.FromBlob(manifestBlob, manifestType)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshall manifest of image : %w", err)
+	}
+	return manifest, nil
+}
+
+func getOCIImgSrcFromPath(ctx context.Context, path string) (types.ImageSource, error) {
+	if !strings.HasPrefix(path, "oci") {
+		path = ociProtocol + path
+	}
+	ociImgRef, err := alltransports.ParseImageName(path)
+	if err != nil {
+		return nil, err
+	}
+	imgsrc, err := ociImgRef.NewImageSource(ctx, nil)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get OCI Image from %s: %w", path, err)
+	}
+	return imgsrc, nil
+}
+
+func getConfigPathFromLabel(imagePath, configSha string) (string, error) {
+	var cfg *manifest.Schema2V1Image
+	configLayerDir := configSha[7:]
+	cfgBlob, err := ioutil.ReadFile(filepath.Join(imagePath, blobsPath, configLayerDir))
+	if err != nil {
+		return "", fmt.Errorf("unable to read the config blob %s from the oci image: %w", configLayerDir, err)
+	}
+	err = json.Unmarshal(cfgBlob, &cfg)
+	if err != nil {
+		return "", fmt.Errorf("problem unmarshaling config blob in %s: %w", configLayerDir, err)
+	}
+	if dirName, ok := cfg.Config.Labels[configsLabel]; ok {
+		return dirName, nil
+	}
+	return "", fmt.Errorf("label %s not found in config blob %s", configsLabel, configLayerDir)
+}
+
+func pullImage(from, to string, srcSkipTLS bool, inStyle style) error {
+	ctx := context.Background()
+	opts := []crane.Option{
+		crane.WithAuthFromKeychain(authn.DefaultKeychain),
+		crane.WithContext(ctx),
+		crane.WithTransport(createRT(srcSkipTLS)),
+	}
+	if srcSkipTLS {
+		opts = append(opts, crane.Insecure)
+	}
+	img, err := crane.Pull(from, opts...)
+	if err != nil {
+		return err
+	}
+	if inStyle == ociStyle {
+		err = crane.SaveOCI(img, to)
+		if err != nil {
+			return fmt.Errorf("unable to save image %s in OCI format in %s: %w", from, to, err)
+		}
+	} else {
+		tags, err := image.GetTagsFromImage(from)
+		tag := "latest"
+		if err == nil || len(tags) > 0 {
+			tag = tags[0]
+		}
+		if err := crane.SaveLegacy(img, tag, to); err != nil {
+			return fmt.Errorf("unable to save image %s in its original format in %s: %w", from, to, err)
+		}
+	}
+	return nil
+}
+
+// newSystemContext set the context for source & destination resources
+func newSystemContext(skipTLS bool) *types.SystemContext {
+	var skipTLSVerify types.OptionalBool
+	if skipTLS {
+		skipTLSVerify = types.OptionalBoolTrue
+	} else {
+		skipTLSVerify = types.OptionalBoolFalse
+	}
+	ctx := &types.SystemContext{
+		RegistriesDirPath:           "",
+		ArchitectureChoice:          "",
+		OSChoice:                    "",
+		VariantChoice:               "",
+		BigFilesTemporaryDir:        "", //*globalArgs.cache + "/tmp",
+		DockerInsecureSkipTLSVerify: skipTLSVerify,
+	}
+	return ctx
+}
+
+// calls the underlying containers/image copy library
+// PS: we could have used crane here is well. Up for reviews
+func copyImage(from, to string, srcSkipTLS bool, dstSkipTLS bool) error {
```

Comment:
I agree it would probably be better to standardize on one registry client for the copy/mirror functionality, if possible.